spec/bolt/node/winrm_spec.rb

```diff
@@ -35,25 +35,26 @@ def stub_winrm_to_raise(klass, message)
   context "when connecting fails", winrm: true do
     it "raises Node::ConnectError if the node name can't be resolved" do
       winrm = Bolt::WinRM.new('totally-not-there', port, user, password)
+      exec_time = Time.now
       expect_node_error(Bolt::Node::ConnectError,
                         'CONNECT_ERROR',
                         /Failed to connect to/) do
         winrm.connect
       end
+      exec_time = Time.now - exec_time
+      expect(exec_time).to be < 1
     end
 
     it "raises Node::ConnectError if the connection is refused" do
       winrm = Bolt::WinRM.new(host, 65535, user, password)
 
-      stub_winrm_to_raise(
-        Errno::ECONNREFUSED,
-        "Connection refused - connect(2) for \"#{host}\" port #{port}"
-      )
-
-      expect_node_error(Bolt::Node::ConnectError,
-                        'CONNECT_ERROR',
-                        /Failed to connect to/) do
-        winrm.connect
+      # timeout will never occur given default 1 second connection timeout
+      Timeout.timeout(2) do
+        expect_node_error(Bolt::Node::ConnectError,
+                          'CONNECT_ERROR',
+                          /Failed to connect to/) do
+          winrm.connect
+        end
```

Comment:
Should probably add a `winrm` test that validates that it gives up after the given `timeout` seconds, like the SSH tests above using netcat.